runtime/src/lib.rs

```diff
@@ -532,139 +554,159 @@ mod parachain_impl {
+        // Xcm origins can be represented natively under the Xcm pallet's Xcm origin.
+        XcmPassthrough<Origin>,
+    );
+
+    parameter_types! {
+        // One XCM operation is 200_000_000 weight, cross-chain transfer ~= 2x of transfer.
+        pub const UnitWeightCost: Weight = 200_000_000;
+        pub KsmPerSecond: (MultiLocation, u128) = (X1(Parent), 100000);
+    }
+
+    pub type Barrier = (
+        TakeWeightCredit,
+        AllowTopLevelPaidExecutionFrom<All<MultiLocation>>,
+    );
+
+    pub struct XcmConfig;
+    impl Config for XcmConfig {
+        type Call = Call;
+        type XcmSender = XcmRouter;
+        type AssetTransactor = LocalAssetTransactor;
+        type OriginConverter = XcmOriginToCallOrigin;
+        type IsReserve = MultiNativeAsset;
+        type IsTeleporter = ();
+        type LocationInverter = LocationInverter<Ancestry>;
+        type Barrier = Barrier;
+        type Weigher = FixedWeightBounds<UnitWeightCost, Call>;
+        type Trader = FixedRateOfConcreteFungible<KsmPerSecond, ()>;
+        type ResponseHandler = ();
+    }
+
+    parameter_types! {
+        pub MaxDownwardMessageWeight: Weight = RuntimeBlockWeights::get().max_block / 10;
+    }
+
+    /// No local origins on this chain are allowed to dispatch XCM sends/executions.
+    pub type LocalOriginToLocation = (SignedToAccountId32<Origin, AccountId, RelayNetwork>,);
+
+    /// The means for routing XCM messages which are not for local execution into the right message
+    /// queues.
+    pub type XcmRouter = (
+        // Two routers - use UMP to communicate with the relay chain:
+        cumulus_primitives_utility::ParentAsUmp<ParachainSystem>,
+        // ..and XCMP to communicate with the sibling chains.
+        XcmpQueue,
+    );
+
+    impl pallet_xcm::Config for Runtime {
+        type Event = Event;
+        type SendXcmOrigin = EnsureXcmOrigin<Origin, LocalOriginToLocation>;
+        type XcmRouter = XcmRouter;
+        type ExecuteXcmOrigin = EnsureXcmOrigin<Origin, LocalOriginToLocation>;
+        type XcmExecuteFilter = All<(MultiLocation, Xcm<Call>)>;
+        type XcmExecutor = XcmExecutor<XcmConfig>;
+        type XcmTeleportFilter = ();
+        type XcmReserveTransferFilter = All<(MultiLocation, Vec<MultiAsset>)>;
+        type Weigher = FixedWeightBounds<UnitWeightCost, Call>;
+    }
+
+    impl cumulus_pallet_xcm::Config for Runtime {
+        type Event = Event;
+        type XcmExecutor = XcmExecutor<XcmConfig>;
+    }
+
+    impl cumulus_pallet_xcmp_queue::Config for Runtime {
+        type Event = Event;
+        type XcmExecutor = XcmExecutor<XcmConfig>;
+        type ChannelInfo = ParachainSystem;
+    }
+
+    impl cumulus_pallet_dmp_queue::Config for Runtime {
+        type Event = Event;
+        type XcmExecutor = XcmExecutor<XcmConfig>;
+        type ExecuteOverweightOrigin = EnsureRoot<AccountId>;
+    }
+
+    pub type LocalAssetTransactor = MultiCurrencyAdapter<
+        NetworkAsset,
```

Comment:
For time being, I have added a new instance of orml-tokens to allow for token transfer to get around the implementation requirement of `orml_traits::MultiCurrency`. Will think of a better solution, either replacing the MultiCurrencyAdapter implementation or supporting pallet_assets in xtokens 🤔

Reply:
`xtokens` doesn't depend on `MultiCurrency`, seems we only need it for this [MultiCurrencyAdapter](https://github.com/open-web3-stack/open-runtime-module-library/blob/master/xcm-support/src/currency_adapter.rs#L44) from `xcm-support`. The code there is simple as it's just a small utility implementing the [`TransactAsset`](https://github.com/paritytech/polkadot/blob/master/xcm/xcm-executor/src/traits/transact_asset.rs#L28) trait. I think is worth creating an `AssetAdapter`/`AssetTransactor` to replace this currency adapter, later we can propose it to the ORML repo and that way we reduce our dependencies.

Reply:
Removed orml-tokens and modified MultiCurrencyAdapter with `Mutate` trait, but not using it to mint/burn right now, that requires more changes wrt CurrencyId/AssetId. For now every xcm deposit will be to UnknownTokens.

Reply:
#121 Will work on this next, don't want to block the demo